### PR TITLE
fix(gatsby-cli,gatsby-recipes): only run recipes when admin is enabled

### PR DESCRIPTION
**Pull request by wardpeet**
## Description

When closing develop sometimes the recipes GraphQL wouldn't exit and stay in limbo. This uses execa to make sure we kill the process. I also added an env var so it only loads with admin.

### CODE REVIEW

File: packages/gatsby-cli/src/create-cli.ts [hunk 1]
```diff
@@ -185,7 +185,12 @@ function buildLocalCommands(cli: yargs.Argv, isLocalSite: boolean): void {
     handler: handlerP(
       getCommandHandler(`develop`, (args: yargs.Arguments, cmd: Function) => {
         process.env.NODE_ENV = process.env.NODE_ENV || `development`
-        startGraphQLServer(siteInfo.directory, true)
+
+        // Only run recipes when admin is enabled
```
Comment (laurieontech) [inline packages/gatsby-cli/src/create-cli.ts:189]: I don't think this will work. The server also needs to run in the CLI when `gatsby recipes` is run.

Comment (wardpeet) [reply packages/gatsby-cli/src/create-cli.ts:189]: This is the develop command doesn't recipes use this one?
https://github.com/gatsbyjs/gatsby/blob/69a93baecb2837c36a67ad8fcdb626c29386ac80/packages/gatsby-cli/src/recipes.ts#L10-L20

Comment (laurieontech) [reply packages/gatsby-cli/src/create-cli.ts:189]: Ah, yes.

**Review by laurieontech — DISMISSED** [review]
There is still the chance I'm missing something here but this should be fine.

**Comment by LekoArts** [comment]
Closing due to https://github.com/gatsbyjs/gatsby/pull/34094

**Pull request closed without merging**